crypto/src/mls/conversation/commit_delay.rs

```diff
@@ -19,57 +19,56 @@ impl MlsConversation {
     pub fn compute_next_commit_delay(&self) -> Option<u64> {
         use openmls::messages::proposals::Proposal;
 
-        if self.group.pending_proposals().count() > 0 {
-            let removed_index = self
-                .group
-                .pending_proposals()
-                .filter_map(|proposal| {
-                    if let Proposal::Remove(remove_proposal) = proposal.proposal() {
-                        Some(remove_proposal.removed())
-                    } else {
-                        None
-                    }
-                })
-                .collect::<Vec<LeafNodeIndex>>();
-
-            let self_index = self.group.own_leaf_index();
-            debug!(removed_index:? = removed_index, self_index:? = self_index; "Indexes");
-            // Find a remove proposal that concerns us
-            let is_self_removed = removed_index.iter().any(|&i| i == self_index);
-
-            // If our own client has been removed, don't commit
-            if is_self_removed {
-                debug!("Self removed from group, no delay needed");
-                return None;
-            }
-
-            let epoch = self.group.epoch().as_u64();
-            let mut own_index = self.group.own_leaf_index().u32() as u64;
-
-            // Look for members that were removed at the left of our tree in order to shift our own leaf index (post-commit tree visualization)
-            let left_tree_diff = self
-                .group
-                .members()
-                .take(own_index as usize)
-                .try_fold(0u32, |mut acc, kp| {
-                    if removed_index.contains(&kp.index) {
-                        acc += 1;
-                    }
-
-                    Result::<_, MlsError>::Ok(acc)
-                })
-                .unwrap_or_default();
-
-            // Post-commit visualization of the number of members after remove proposals
-            let nb_members = (self.group.members().count() as u64).saturating_sub(removed_index.len() as u64);
-            // This shifts our own leaf index to the left (tree-wise) from as many as there was removed members that have a smaller leaf index than us (older members)
-            own_index = own_index.saturating_sub(left_tree_diff as u64);
-
-            Some(Self::calculate_delay(own_index, epoch, nb_members))
-        } else {
+        if self.group.pending_proposals().count() == 0 {
```

Comment:
Strictly we don't even need a full count here; `self.group.pending_proposals().next().is_none()` avoids unnecessary iteration.